src/modules/python/pbs/v1/_svr_types.py

```diff
@@ -1365,3 +1365,136 @@ def __next__(self):
                 return _pbs_v1.iter_nextfunc(self, 0, self.obj_name,
                                              self.filter1, self.filter2)
 #: C(pbs_iter)
+
+#:------------------------------------------------------------------------
+#                  SERVER ATTRIBUTE TYPE
+#:-------------------------------------------------------------------------
+class _server_attribute:
+    """
+    This represents a external form of attributes..
+    """
+    attributes = PbsReadOnlyDescriptor('attributes', {})
+    _attributes_hook_set = {}
+    def __init__(self, name, resource, value, op, flags):
+        self.name = name
+        self.resource = resource
+        self.value = value
+        self.op = op
+        self.flags = flags
+        self.sisters = []
+    #: m(__init__)
+
+    def __str__(self):
+        return "%s:%s:%s:%s:%s" % self.tup()
+    #: m(__str__)
+
+    def __setattr__(self, name, value):
+        if _pbs_v1.in_python_mode():
+            raise BadAttributeValueError(
+                "'%s' attribute in the server_attribute object is readonly" % (name,))
+        super().__setattr__(name, value)
+    #: m(__setattr__)
+
+    def extract_flags_str(self):
+        """returns the string values from the attribute flags."""
+        lst = []
+        for mask, value in _pbs_v1.REVERSE_ATR_VFLAGS.items():
+            if self.flags & mask:
+                lst.append(value)
+        return lst
+    #: m(extract_flags_str)
+
+    def extract_flags_int(self):
+        """returns the integer values from the attribute flags."""
+        lst = []
+        for mask, value in _pbs_v1.REVERSE_ATR_VFLAGS.items():
+            if self.flags & mask:
+                lst.append(mask)
+        return lst
+    #: m(extract_flags_int)
+
+    def tup(self):
+        return self.name, self.resource, self.value, self.op, self.flags, self.sisters
+    #: m(tup)
+
+_server_attribute._connect_server = PbsAttributeDescriptor(
+    _server_attribute, '_connect_server', "", (str,))
```

Comment:
Do we need these lines?  What purpose do they server?

Reply:
It does look like it may not have any use right now, but perhaps in the future. It sort of associates the current class object with the particular PBS server. There could be multiple servers in the future being handled in hooks. So I'd say just leave it for now.

Reply:
I will leave it.